students/rhamersky/Lesson_3/mailroom_part1.py

```diff
@@ -0,0 +1,269 @@
+        return
+
+    @staticmethod
+    def create_report():
+        '''
+        This method takes the table tuple and creates a summary report for the user (name, number of gifts, total amount
+        in gifts, and the average amount given. The code sorts the list rows into the largest amount amount give to the
+        least amount given.
+        :return: This method does not return anything. Return is just used to clearly show where this method ends.
+        '''
+        total = int()
+        donation_count = int()
+        lst_table = []
+        # Creates a new list [name, number of gifts, average amount given, total amount in gifts] to be used to create
+        # the summary chart.
+        for row in tup_donor_names:
+            for value in range(2, len(row)):
+                total += row[value]
+                donation_count += 1
+            lst_table.append([row[0], row[1], total/donation_count, total])
+            total = 0
+            donation_count = 0
+        # Reverses the list
+        for row in lst_table:
+            row.reverse()
+        # Shorts the list form largest to smallest on the first element of each row.
+        lst_table.sort(reverse=True)
+
+        index = 0
+        # Formatting the money elements and changing all int() & float() data to str() data
+        # to use for creating the chart.
+        for row in range(len(lst_table)):
+            lst_table[row][index] = "${:,.2f}".format(lst_table[row][index])  # --> Total amount for gifts
+            lst_table[row][index + 1] = "${:,.2f}".format(lst_table[row][index + 1])  # --> Avg. gift amount
+            lst_table[row][index + 2] = str(lst_table[row][index + 2])  # --> Number of gifts
+        # Reverses list
+        for row in lst_table:
+            row.reverse()
+
+        lst_headers = ["Donor_Name", "Num_Gifts", "Average_Gift", "Total_Given"]  # --> List of headers for the chart.
+        # Calculating the max length of all the elements.
+        length1 = max(len(word) for word in lst_headers) + 10  # --> Max length for the header list
+        length2 = max(len(word) for row in lst_table for word in row)  # --> Max length for the donor table
+        # Calculates the max between the header list and the donor list
+        max_length = max(length1, length2)
+        # Creates the header for the chart
+        for heading in lst_headers:
+            print(("{0:{padding}}".format(heading, padding=max_length)), end=" ")
+        print()
+        # Creates the donor rows for the chart.
+        for row in lst_table:
+            print(("{0:{padding}}".format(row[0], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[1], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[2], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[3], padding=max_length)), end=" ")
+            print()
+        return
+
+
+def donor_history(donor_name, donation_amount):
```

Comment:
it is important for function names to closely reflect what it does, in this case this should be something like `update_donors`

`donor_history` is a bit misleading